cmd/deploy/deploy.go

```diff
@@ -262,6 +264,7 @@ func Deploy(ctx context.Context) *cobra.Command {
 	cmd.Flags().StringVarP(&options.Namespace, "namespace", "n", "", "overwrites the namespace where the development environment is deployed")
 	cmd.Flags().StringVarP(&options.K8sContext, "context", "c", "", "context where the development environment is deployed")
 	cmd.Flags().StringArrayVarP(&options.Variables, "var", "v", []string{}, "set a variable (can be set more than once)")
+	cmd.Flags().StringVarP(&options.SshIdentityKey, "identity-file", "i", "", "the identity key to use for ssh sessions")
```

Comment:
Why do we need a new flag? Could we use the one locally and pass it through an arg to the remote?

 I think adding a new flag with a shortcut that is only going to be used on this command can led to future problems. WDYT?

Reply:
Users should be able to provide a different identity key to authenticate:

```bash
okteto deploy --remote -i "$HOME/.ssh/id_rsa"
```

I agree it's confusing since this flag will only work with `--remote` and we should probably add a guard for that and fail if it's not provided.

Can you think of a different way of achieving/doing this?

Reply:
`-i "$HOME/.ssh/id_rsa"` would probably break replicability between developers.
My 2$ is to ship the first version without it. A workaround, if someone needs this, is to add the private image to the deploy `image` itself. But my understanding is that the ssh agent should cover most of the use cases

Reply:
I don't know if we are using it in any other place rather than the creation of the Dockerfile, at least is the only place I can see it. 

We can always use the environment variables/args/secrets to inject things into buildkit to skip adding the flag